src/main/java/org/dataone/hashstore/filehashstore/FileHashStoreUtility.java

```diff
@@ -0,0 +1,117 @@
+            throw new NullPointerException(errMsg);
+        }
+    }
+
+    /**
+     * Calculate the hex digest of a pid's respective object with the given algorithm
+     *
+     * @param dataStream InputStream to object
+     * @param algorithm  Hash algorithm to use
+     * @return Hex digest of the pid's respective object
+     * @throws IOException              Error when calculating hex digest
+     * @throws NoSuchAlgorithmException Algorithm not supported
+     */
+    public static String calculateHexDigest(InputStream dataStream, String algorithm)
+        throws IOException, NoSuchAlgorithmException {
+        MessageDigest mdObject = MessageDigest.getInstance(algorithm);
+        try {
+            byte[] buffer = new byte[8192];
+            int bytesRead;
+            while ((bytesRead = dataStream.read(buffer)) != -1) {
+                mdObject.update(buffer, 0, bytesRead);
+
+            }
+            // Close dataStream
+            dataStream.close();
+
+        } catch (IOException ioe) {
+            String errMsg =
+                "FileHashStoreUtility.calculateHexDigest - Unexpected IOException encountered: "
+                    + ioe.getMessage();
+            throw new IOException(errMsg);
+
+        }
+        // mdObjectHexDigest
+        return DatatypeConverter.printHexBinary(mdObject.digest()).toLowerCase();
+    }
+
+    /**
+     * Checks whether a directory is empty or contains files. If a file is found, it returns true.
+     *
+     * @param directory Directory to check
+     * @return True if a file is found or the directory is empty, False otherwise
+     * @throws IOException If I/O occurs when accessing directory
+     */
+    public static boolean isDirectoryEmpty(Path directory) throws IOException {
+        try (Stream<Path> stream = Files.list(directory)) {
+            // The findFirst() method is called on the stream created from the given
+            // directory to retrieve the first element. If the stream is empty (i.e., the
+            // directory is empty), findFirst() will return an empty Optional<Path>.
+            //
+            // The isPresent() method is called on the Optional<Path> returned by
+            // findFirst(). If the Optional contains a value (i.e., an element was found),
+            // isPresent() returns true. If the Optional is empty (i.e., the stream is
+            // empty), isPresent() returns false.
+            return !stream.findFirst().isPresent();
+        }
+    }
+
+    /**
+     * Checks whether a given string is empty and throws an exception if so
+     *
+     * @param string   String to check
+     * @param argument Value that is being checked
+     * @param method   Calling method
+     */
+    public static void checkForEmptyString(String string, String argument, String method) {
+        if (string.trim().isEmpty()) {
+            String errMsg = "FileHashStoreUtility.isStringNullOrEmpty - Calling Method: " + method
```

Comment:
Apologies for the confusion - previously I combined the check for checking nulls and for an empty string, but have separated it since and missed updating the exception messaging here (and in your other comments).